[WFLY-19442]: [DOCS]  Adding documentation covering the umanaged deployments

ehsavoie opened the pull request:
* Adding documentation

Jira: https://issues.redhat.com/browse/WFCORE-6503

____

<--- THIS SECTION IS AUTOMATICALLY GENERATED BY WILDFLY GITHUB BOT. ANY MANUAL CHANGES WILL BE LOST. --->

> Wildfly issue links:
> * [WFLY-19442](https://issues.redhat.com/browse/WFLY-19442)

<--- END OF WILDFLY GITHUB BOT REPORT --->

More information about the [wildfly-bot[bot]](https://github.com/wildfly/wildfly-github-bot)

mnovak reviewed (approved):
LGTM

bstansberry commented on docs/src/main/asciidoc/_admin-guide/management-tasks/YAML_Configuration_file.adoc:
```diff
@@ -107,3 +107,33 @@ wildfly-configuration:
+ - path: since those should already have been defined when the YAML files are parsed.
+
+While *managed* deployments are not supported you can still define *unmanaged* deployments that are not meant to be managed using the management API which is fine since you are in *read-only* mode with YAML and your changes aren't persisted.
+This is a sample to install two deployments but only one is deployed:
```
s/but only one is/but with only one/g

bstansberry commented on docs/src/main/asciidoc/_admin-guide/management-tasks/YAML_Configuration_file.adoc:
```diff
@@ -107,3 +107,33 @@ wildfly-configuration:
+          archive: true
+----
+
+NOTE: The `content` attribute is of type LIST so you must add a *-* before defining the archive or the directory of your deployment.
```
Perhaps

s/The/For historical reasons, the/g

and then add a sentence

'The list should only contain a single element.'

bstansberry commented on docs/src/main/asciidoc/_admin-guide/management-tasks/YAML_Configuration_file.adoc:
```diff
@@ -107,3 +107,33 @@ wildfly-configuration:
+ - deployment-overlay: to add deployment-overlays to the server as this requires more than just some configuration.
+ - path: since those should already have been defined when the YAML files are parsed.
+
+While *managed* deployments are not supported you can still define *unmanaged* deployments that are not meant to be managed using the management API which is fine since you are in *read-only* mode with YAML and your changes aren't persisted.
```
This isn't truly accurate. These resources can still be manipulated using the management API. It's true that changes are not persisted but that's not the same thing.

While *managed* deployments are not supported you can still define *unmanaged* deployments. An unmanaged deployment is one where the server does not make an internal copy of the deployment content for its use; instead it directly uses the configured content. See [Managed and Unmanaged Deployments] (https://docs.wildfly.org/32/Admin_Guide.html#managed-and-unmanaged-deployments) for details.

except a proper adoc ref link instead of my lazy github ui syntax ;)

yersan reviewed (approved):
LGTM

yersan commented:
@bstansberry your review was already addressed, since this is a documentation issue and AD was approved by all the parties, I merged the related RFE but kept this for your final review

bstansberry commented on docs/src/main/asciidoc/_admin-guide/management-tasks/YAML_Configuration_file.adoc:
```diff
@@ -107,3 +107,36 @@ wildfly-configuration:
+
+While *managed* deployments are not supported you can still define *unmanaged* deployments. An unmanaged deployment is one where the server does not make an internal copy of the deployment content for its use; instead it directly uses the configured content. See link:#managed-and-unmanaged-deployments[Managed and Unmanaged Deployments].
+
+IMPORTANT: All your changes using the management management API aren't persisted since the server is in *read-only* mode with YAML.
```
s/management management/management

ehsavoie committed:
[WFLY-19442]: Adding documentation covering the umanaged deployments

 * Adding documentation

Jira: https://issues.redhat.com/browse/WFLY-19442
      https://issues.redhat.com/browse/WFCORE-6503
Signed-off-by: Emmanuel Hugonnet <ehugonne@redhat.com>

bstansberry commented:
Thanks @ehsavoie
